Paper draft for submission at JOSS (#272)
mne-tools/mne-bids: paper/paper.md

---
title: 'MNE-BIDS'
tags:
  - Python
  - BIDS
  - MNE
  - neuroimaging
  - MEG
  - EEG

> authors:

Reviewer: missing many authors. I'm assuming it's still WIP

Reviewer: maybe tag the authors so that they can give you the affiliation in "suggesting" mode

Author: > missing many authors. I'm assuming it's still WIP

indeed, see the "What needs to be done" heading in my OP

But good call, I'll tag people.

Author: done, see my updated OP

  - affiliation: 1
    name: Stefan Appelhoff
    orcid: 0000-0001-8002-0877
  - affiliation: 2
    name: Mainak Jas
    orcid: 0000-0002-3199-9027    
  - affiliation: 3
    name: Chris Holdgraf
    orcid: 0000-0002-2391-0678
  - affiliation:
    name:
    orcid:
affiliations:
- index: 1
  name: Center for Adaptive Rationality, Max Planck Institute for Human Development, Berlin, Germany
- index: 2
  name: Télécom ParisTech, France  
- index: 3
  name: University of California at Berkeley
- index:
  name:
date: 25 October 2019
bibliography: paper.bib
---

# Summary

With the emergence of the Brain Imaging Data Structure
(``BIDS``; [@Gorgolewski2016]) in 2016, the scientific community received a
standard to organize and share data in the broad domain of neuroscience.

Originally limited to MRI data types, BIDS is continuously being extended and
by now also supports other neuroimaging modalities such as MEG [@Niso2018] and

> EEG [@Pernet2019].

Reviewer: ```suggestion
EEG [@Pernet2019] and iEEG [...].
```

BIDS prescribes how complex experiment data should be structured and which
metadata should be encoded next to the raw data. This set of guidelines
allows users to reap several benefits, such as:

1. **sharing data within a lab and between labs:** Through reading the BIDS
   specification, users are empowered to understand any BIDS dataset
   without requiring specialist knowledge.
1. **data validation:** Dedicated software can check whether the rules of
   BIDS are being followed in a dataset, and alert users to missing or broken
   data (see

>    [bids-validator](https://github.com/bids-standard/bids-validator)).

Reviewer: I think there should be some statement about the fact that testing is done with the validator so that we guarantee the outputs are BIDS compliant

Author: Thanks! I added a sentence for this.

``MNE-BIDS`` supports conversion from raw data formats that are not BIDS
compatible into permissible formats. As a result, users can easily convert
their datasets to BIDS in a matter of minutes, rather than after hours of
manual labour.

Next to this core functionality, ``MNE-BIDS`` is continuously being extended
to facilitate the analysis of BIDS formatted data. To name some features, it is
possible to read a raw BIDS dataset and obtain a Python object, ready for
analyis with MNE-Python, or to save a T1 weighted anatomical MRI image
alongside the MEG or EEG data and apply an automatic defacing algorithm for
anonymization purposes

Users can easily install ``MNE-BIDS`` on all platforms via `pip` and `conda`
and its functionality is continuously tested on Windows and Linux.
Next to three core dependencies for scientific computation (`numpy`, `scipy`),
and handling of MEG and EEG data (`mne`), ``MNE-BIDS`` has only minimal
dependencies, all of which are optional. The API of the package is stable and
extensively documented and explained in examples (https://mne.tools/mne-bids/).

As of writing, ``MNE-BIDS`` has received code contributions from 15

> contributors and its' user base is steadily growing. Code development is

Reviewer: ```suggestion
contributors and its user base is steadily growing. Code development is
```

Author: but it's the user base of the package (it) ... so its' --> this should be correct.

Reviewer: not sure I follow. The possessive "its" does not have an apostrophe unlike for example "Mainak's" or "Stefan's"

Author: haha, okay I checked this thoroughly now ... what I suggested does not exist. Thanks :-D

---
title: 'MNE-BIDS'
tags:
  - Python
  - BIDS
  - MNE
  - neuroimaging
  - MEG

>   - EEG

Reviewer 2: ```suggestion
  - EEG
  - iEEG
```

authors:
  - affiliation: 1
    name: Stefan Appelhoff
    orcid: 0000-0001-8002-0877
  - affiliation: 2
    name: Mainak Jas
    orcid: 0000-0002-3199-9027
  - affiliation: 3
    name: monkeyman192
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 4
    name: Teon Brooks

>     orcid: xxxx-xxxx-xxxx-xxxx

Reviewer 2: ```suggestion
    orcid: 0000-0001-7344-3230
```

  - affiliation: 5
    name: wmvanvliet
    orcid:
  - affiliation: 6
    name: romquentin
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 7
    name: Chris Holdgraf
    orcid: 0000-0002-2391-0678
  - affiliation: 8
    name: dnacombo
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 9
    name: ezemikulan
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 10
    name: ktavabi
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 11
    name: hoechenberger
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 12
    name: dominikwelke
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 13
    name: Clemens Brunner
    orcid: 0000-0002-6030-2233
  - affiliation: 14
    name: alexrockhill
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 15
    name: Alexandre Gramfort
    orcid: xxxx-xxxx-xxxx-xxxx
affiliations:
- index: 1
  name: Center for Adaptive Rationality, Max Planck Institute for Human Development, Berlin, Germany
- index: 2
  name: Télécom ParisTech, France
- index: 3
  name:
- index: 4

>   name:

Reviewer 2: ```suggestion
  name: Mozilla
```

standard to organize and share data in the broad domain of neuroscience.

Originally limited to MRI data types, BIDS is continuously being extended and
by now also supports other neuroimaging modalities such as MEG [@Niso2018],
EEG [@Pernet2019], and iEEG [@Holdgraf2019].

BIDS prescribes how complex experiment data should be structured and which
metadata should be encoded next to the raw data. This set of guidelines
allows users to reap several benefits, such as:

1. **sharing data within a lab and between labs:** Through reading the BIDS
   specification, users are empowered to understand any BIDS dataset
   without requiring specialist knowledge.
1. **data validation:** Dedicated software can check whether the rules of
   BIDS are being followed in a dataset, and alert users to missing or broken
   data (see
   [bids-validator](https://github.com/bids-standard/bids-validator)).
1. **automatic data handling and analysis pipelines:** With a perfect
   knowledge about which data to expect and where to search for it, many
   common workflows in data handling and analysis can be simplified or
   automated through specialized software.

> To make use of these benefits however, datasets first must be converted to

Reviewer 2: ```suggestion
To make use of these benefits however, datasets must first be converted to
```

BIDS format. Furthermore, already existing data analysis software must be
extended to be aware of a BIDS format and how to properly use it.

This is where ``MNE-BIDS`` steps in: Harnessing BIDS and the data analysis
software MNE-Python [@Agramfort2013], it is the goal of ``MNE-BIDS`` to

> automatically organize raw MEG and EEG data into BIDS format and to facilitate

Reviewer 2: ```suggestion
automatically organize raw MEG, EEG, and iEEG data into BIDS format and to facilitate
```

their analysis.

> Starting with, for example, a single directory full of datafiles with arbitrary

Reviewer 2: ```suggestion
For example, starting with a single directory full of datafiles with arbitrary
```

> naming, ``MNE-BIDS`` can be used to extract present meta data, reorganize the

Reviewer 2: ```suggestion
naming, ``MNE-BIDS`` can be used to extract present metadata, reorganize the
```

> files into BIDS format, and write additional meta data. Moreover,

Reviewer 2: ```suggestion
files into BIDS format, and write additional metadata. Moreover,
```

``MNE-BIDS`` supports conversion from raw data formats that are not BIDS
compatible into permissible formats. As a result, users can easily convert
their datasets to BIDS in a matter of minutes, rather than after hours of
manual labour. All conversions performed by MNE-BIDS are validated with the
bids-validator for a maximum certainty that the results hold up to the BIDS
format.

Next to this core functionality, ``MNE-BIDS`` is continuously being extended
to facilitate the analysis of BIDS formatted data. To name some features, it is
possible to read a raw BIDS dataset and obtain a Python object, ready for
analyis with MNE-Python, or to save a T1 weighted anatomical MRI image
alongside the MEG or EEG data and apply an automatic defacing algorithm for

> anonymization purposes

Reviewer 2: ```suggestion
anonymization purposes.
```

Users can easily install ``MNE-BIDS`` on all platforms via `pip` and `conda`
and its functionality is continuously tested on Windows and Linux.
Next to three core dependencies for scientific computation (`numpy`, `scipy`),

> and handling of MEG and EEG data (`mne`), ``MNE-BIDS`` has only minimal

Reviewer 2: ```suggestion
and handling of MEG, EEG, and iEEG data (`mne`), ``MNE-BIDS`` has minimal
```

   knowledge about which data to expect and where to search for it, many
   common workflows in data handling and analysis can be simplified or
   automated through specialized software.

To make use of these benefits, datasets must first be converted to
BIDS format. Furthermore, already existing data analysis software must be
extended to be aware of a BIDS format and how to properly use it.

This is where ``MNE-BIDS`` steps in: Harnessing BIDS and the data analysis
software MNE-Python [@Agramfort2013], it is the goal of ``MNE-BIDS`` to
automatically organize raw MEG, EEG, and iEEG data into BIDS format and to facilitate
their analysis.

For example, starting with a single directory full of datafiles with arbitrary
naming, ``MNE-BIDS`` can be used to extract present metadata, reorganize the
files into BIDS format, and write additional metadata. Moreover,
``MNE-BIDS`` supports conversion from raw data formats that are not BIDS
compatible into permissible formats. As a result, users can easily convert
their datasets to BIDS in a matter of minutes, rather than after hours of
manual labour. All conversions performed by MNE-BIDS are validated with the

> bids-validator for a maximum certainty that the results hold up to the BIDS

Reviewer 3: CAN we link to validator?

Author: it's linked above, but I will insert another link as well. thanks!

format.

Next to this core functionality, ``MNE-BIDS`` is continuously being extended
to facilitate the analysis of BIDS formatted data. To name some features, it is
possible to read a raw BIDS dataset and obtain a Python object, ready for
analyis with MNE-Python, or to save a T1 weighted anatomical MRI image
alongside the MEG or EEG data and apply an automatic defacing algorithm for

> anonymization purposes.

Reviewer 3: Say also the fiducials points can be saved to store coregistration info that allows automatic computation of forward and inverse solutions

Users can easily install ``MNE-BIDS`` on all platforms via `pip` and `conda`

> and its functionality is continuously tested on Windows and Linux.

Reviewer 3: Not Mac ? We should if we don’t

Author: we have:
- circleci for building docs (running on linux)
- travis CI for testing (linux)
- appveyor for testing (windows)

I agree that testing on mac would be nice. I'll open an issue for this.

Next to three core dependencies for scientific computation (`numpy`, `scipy`),
and handling of MEG, EEG, and iEEG data (`mne`), ``MNE-BIDS`` has minimal
dependencies, all of which are optional. The API of the package is stable and
extensively documented and explained in examples (https://mne.tools/mne-bids/).

As of writing, ``MNE-BIDS`` has received code contributions from 15
contributors and its user base is steadily growing. Code development is
active and users are typically receiving support within a few hours of opening
an issue on our dedicated issue tracker.

MNE-BIDS is used in automated analysis pipelines such as the

> MNE-study-template (https://github.com/mne-tools/mne-study-template) and

Reviewer 3: I would say that this template emerged from Mainak frontiers paper

several large institutions such as Martinos and "insert Matt's institution" have started to use MNE-BIDS.

The developer team is excited to improve the state of the art in data handling
and looking forward to welcoming new contributors and users.

# Acknowledgements

> MNE-BIDS development is partly supported by XYZ.

Reviewer 3: Add French ANR grant and NIH funding in the US

Author: Is this a note to yourself? :-) I don't know which specific lines to include.

As for my funding, I am in touch with @franklin-feingold to figure out how I can acknowledge the funding from the US that I receive via a grant of Russ

Reviewer 4: My contributions to MNE-BIDS are funded by the Academy of Finland, grant 310988.

    orcid: 0000-0001-9791-4404
affiliations:
- index: 1
  name: Center for Adaptive Rationality, Max Planck Institute for Human Development, Berlin, Germany
- index: 2
  name: Athinoula A. Martinos Center for Biomedical Imaging, Massachusetts General Hospital, Charlestown, MA, USA
- index: 3
  name:
- index: 4
  name: Mozilla
- index: 5
  name: Department of Neuroscience and Biomedical Engineering, Aalto University, Espoo, Finland
- index: 6
  name: Human Cortical Physiology and Neurorehabilitation Section, NINDS, NIH, Bethesda, Maryland 20892
- index: 7
  name: UC Berkeley, Project Jupyter
- index: 8
  name:
- index: 9
  name: Department of Biomedical and Clinical Sciences 'L. Sacco', University of Milan, Milan, Italy
- index: 10
  name:
- index: 11
  name:
- index: 12
  name:
- index: 13
  name: Institute of Psychology, University of Graz, Austria
- index: 14
  name: University of Oregon, Eugene OR, USA
- index: 15
  name: Université Paris-Saclay, Inria, CEA, Palaiseau, France
date: 28 October 2019
bibliography: paper.bib
---

# Summary

With the emergence of the Brain Imaging Data Structure
(``BIDS``; [@Gorgolewski2016]) in 2016, the scientific community received a

> standard to organize and share data in the broad domain of neuroscience.

Reviewer 5: ... the neuroscientific community received a standard to organize and share data.

> Originally limited to MRI data types, BIDS is continuously being extended and

Reviewer 5: Combine this paragraph with the previous one.
data types -> data formats

by now also supports other neuroimaging modalities such as MEG [@Niso2018],
EEG [@Pernet2019], and iEEG [@Holdgraf2019].

> BIDS prescribes how complex experiment data should be structured and which

Reviewer 5: experiment -> experimental

> metadata should be encoded next to the raw data. This set of guidelines

Reviewer 5: next to -> besides

> metadata should be encoded next to the raw data. This set of guidelines

Reviewer 4: `encoded` -> `present`

> allows users to reap several benefits, such as:

Reviewer 5: ~~allows users to reap several benefits, such as:~~
provides the following benefits:

Reviewer 5: Then I'd capitalize the first letter of each bullet point.

1. **sharing data within a lab and between labs:** Through reading the BIDS

>    specification, users are empowered to understand any BIDS dataset

Reviewer 5: The BIDS specification empowers users to understand...

   without requiring specialist knowledge.
1. **data validation:** Dedicated software can check whether the rules of

>    BIDS are being followed in a dataset, and alert users to missing or broken

Reviewer 5: remove the comma

   data (see
   [bids-validator](https://github.com/bids-standard/bids-validator)).
1. **automatic data handling and analysis pipelines:** With a perfect

>    knowledge about which data to expect and where to search for it, many

Reviewer 5: Knowing which data...

   common workflows in data handling and analysis can be simplified or

>    automated through specialized software.

Reviewer 5: software tools

> To make use of these benefits, datasets must first be converted to

Reviewer 5: to the

BIDS format. Furthermore, already existing data analysis software must be

> extended to be aware of a BIDS format and how to properly use it.

Reviewer 5: a -> the

> This is where ``MNE-BIDS`` steps in: Harnessing BIDS and the data analysis

Reviewer 4: `:` -> `.`

or

`Harnessing` -> `harnessing`

> This is where ``MNE-BIDS`` steps in: Harnessing BIDS and the data analysis

Reviewer 5: ... and the Python-based data analysis toolbox MNE-Python...

Reviewer 4: -1 for 'Python-based data analysis toolbox', I find just 'data analysis software' more elegant. 'Python' is already in the 'MNE-Python' name.

Author: I left it with "data analysis software" for now, but will leave this comment open for others to tip the scales

software MNE-Python [@Agramfort2013], it is the goal of ``MNE-BIDS`` to

> automatically organize raw MEG, EEG, and iEEG data into BIDS format and to facilitate

Reviewer 5: data -> datasets
...into the BIDS format...

their analysis.

> For example, starting with a single directory full of datafiles with arbitrary

Reviewer 5: data files

> naming, ``MNE-BIDS`` can be used to extract present metadata, reorganize the

Reviewer 5: naming -> names
present -> existing

> files into BIDS format, and write additional metadata. Moreover,

Reviewer 5: the BIDS format

> ``MNE-BIDS`` supports conversion from raw data formats that are not BIDS

Reviewer 5: ...supports converting raw data formats...

compatible into permissible formats. As a result, users can easily convert

> their datasets to BIDS in a matter of minutes, rather than after hours of

Reviewer 4: remove `after`

manual labour. All conversions performed by MNE-BIDS are validated with the

> [bids-validator](https://github.com/bids-standard/bids-validator) for a maximum

Reviewer 5: ...are automatically validated with the bids-validator. (remove the rest of the sentence)

Author: I changed this sentence based on a feedback by Marijn and it seems to end better now, so I won't remove the end of the sentence.

Reviewer 5: I still think "validated" is enough, no need to explain what that means. But feel free to leave it in, I'm just making suggestions.

Author: yep, thanks a lot - great suggestions !

> certainty that the results hold up to the BIDS format.

Reviewer 4: `or a maximum certainty that the results hold up to the BIDS format.` ->
`to ensure compatibility with the BIDS format.`

> Next to this core functionality, ``MNE-BIDS`` is continuously being extended

Reviewer 5: Next to -> Besides or In addition

to facilitate the analysis of BIDS formatted data. To name some features, it is
possible to read a raw BIDS dataset and obtain a Python object, ready for

> analyis with MNE-Python. Users can save a T1 weighted anatomical MRI image

Reviewer 5: T1-weighted

> alongside the MEG or EEG data and apply an automatic defacing algorithm for

Reviewer 5: Do you have a link/reference for the defacing algorithm? This might be interesting for a lot of readers.

Reviewer 3: Also we should mention somewhere that it allows to anonymize the data by removing HIS data and applying time shifts

Author: @alexrockhill I think you implemented this feature, right? Can you address this question, please?

anonymization purposes. As a last example, fiducial points can be saved to

> store coregistration information between the electrophysiology and mri data,

Reviewer 5: mri -> MRI

which allows for automatic computation of forward and inverse solutions.

Users can easily install ``MNE-BIDS`` on all platforms via `pip` and `conda`

> and its functionality is continuously tested on Windows and Linux.

Reviewer 5: insert an Oxford comma before and

> Next to three core dependencies for scientific computation (`numpy`, `scipy`),

Reviewer 4: `Next to` -> `Other than`

> Next to three core dependencies for scientific computation (`numpy`, `scipy`),

Reviewer 5: Next to -> Besides

Reviewer 5: computation -> computing

Reviewer 5: remove the comma after (`numpy`, `scipy`)

> and handling of MEG, EEG, and iEEG data (`mne`), ``MNE-BIDS`` has minimal

Reviewer 5: maybe replace MEG, EEG, and iEEG with neurophysiological? no need to be so specific here IMO

dependencies, all of which are optional. The API of the package is stable and
extensively documented and explained in examples (https://mne.tools/mne-bids/).

As of writing, ``MNE-BIDS`` has received code contributions from 15
contributors and its user base is steadily growing. Code development is
active and users are typically receiving support within a few hours of opening

> an issue on our dedicated issue tracker.

Reviewer 5: As someone has already noted, these statements (active development, growing user base, support response time) should be backed up with actual facts.

Reviewer 4: Or maybe just removed? It reads a bit too much like advertising for my taste.

Author: I guess the facts are openly visible on the GitHub page. Were you thinking about some summary statistics @cbrnr ? If so, which ones, and how to compile them?

@wmvanvliet - yes, this is advertising a bit. But we are writing about software, not about science - so I feel like it has a place here :-)

Reviewer 5: Summary statistics would be nice, but I have no idea where to get them without too much work. So maybe at least mention some example numbers to quantify what you mean by e.g. "code development is active" (you could insert the data of the latest commit or something). Or maybe rephrase to "the development team is committed to provide support within only a few hours of opening an issue".  I don't know, I just feel like you need to throw in a few numbers with your claims, but maybe that's just me.

Author: I revised this section, adding a link, making it less advertising, and adding Clemens' sentence

MNE-BIDS is used in automated analysis pipelines such as the
MNE-study-template (https://github.com/mne-tools/mne-study-template)
[@Mainak2018] several large institutions such as

> Martinos and "insert Matt's institution" have started to use MNE-BIDS.

Reviewer 5: This paragraph should be revised.

Author: Right, we need:

1. info on how to refer to Martinos officially. @jasmainak can you help please?
1. info about Matt's institution (@monkeyman192 ) - I already asked @teonbrooks in https://github.com/mne-tools/mne-bids/pull/272#issuecomment-547324591 whether he knows how to contact Matt in case he does not see all of this.

Reviewer: It's Macquarie university for Matt and I'd say you can refer to "Massachussetts General Hospital" or "Martinos Center for Biomedical Imaging"

tags:
  - Python
  - BIDS
  - MNE
  - neuroimaging
  - MEG
  - EEG
  - iEEG
authors:
  - affiliation: 1
    name: Stefan Appelhoff
    orcid: 0000-0001-8002-0877
  - affiliation: 2
    name: Mainak Jas
    orcid: 0000-0002-3199-9027
  - affiliation: 3
    name: monkeyman192
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 4
    name: Teon L Brooks
    orcid: 0000-0001-7344-3230
  - affiliation: 5
    name: Marijn van Vliet
    orcid: 0000-0002-6537-6899
  - affiliation: 6
    name: Romain Quentin
    orcid: 0000-0001-7659-3605
  - affiliation: 7
    name: Chris Holdgraf
    orcid: 0000-0002-2391-0678
  - affiliation: 8
    name: Maximilien Chaumon
    orcid: 0000-0001-9664-8861
  - affiliation: 9
    name: Ezequiel Mikulan
    orcid: 0000-0001-7259-6120
  - affiliation: 10
    name: ktavabi
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 11
    name: hoechenberger
    orcid: xxxx-xxxx-xxxx-xxxx
  - affiliation: 12
    name: Dominik Welke
    orcid: 0000-0002-5529-1998
  - affiliation: 13
    name: Clemens Brunner
    orcid: 0000-0002-6030-2233
  - affiliation: 14
    name: Alexander P Rockhill
    orcid: 0000-0003-3868-7453
  - affiliation: 15
    name: Alexandre Gramfort

>     orcid: 0000-0001-9791-4404

Reviewer: @sappelhoff you should check your google hangouts :) anyway I'll say it here. Any objection to moving me here?

Also add @larsoner - he made significant contributions I'd say

Reviewer 6: Fine by me if you think it's warranted -- Eric Larson, 0000-0003-4782-5360, Institute for Learning and Brain Sciences, University of Washington, Seattle, WA, USA

- index: 7
  name: Institut du cerveau et de la moelle épinière (ICM), Paris, France
- index: 8
  name: Department of Biomedical and Clinical Sciences 'L. Sacco', University of Milan, Milan, Italy
- index: 9
  name: Institute for Learning and Brain Sciences, University of Washington, Seattle, WA, USA
- index: 10
  name: Institute of Neuroscience and Medicine (INM-3), Research Center Jülich, Germany
- index: 11
  name: Max-Planck-Institute for Empirical Aesthetics, Frankfurt a.M., Germany
- index: 12
  name: Institute of Psychology, University of Graz, Austria
- index: 13
  name: University of Oregon, Eugene OR, USA
- index: 14
  name: Université Paris-Saclay, Inria, CEA, Palaiseau, France
- index: 15
  name: Athinoula A. Martinos Center for Biomedical Imaging, Massachusetts General Hospital, Charlestown, MA, USA
date: 29 October 2019
bibliography: paper.bib
---

# Summary

With the development of the Brain Imaging Data Structure
(``BIDS``; [@Gorgolewski2016]), the neuroscientific community received
a standard to organize and share data. BIDS standardizes storage of complex experimental
data by prescribing a set of file naming conventions and templates to store the

> associated metadata in the form of Javascript Object Notation (JSON) files. It thus

Reviewer 4: ```suggestion
associated metadata in the form of Javascript Object Notation (JSON) and tab separated value (TSV) files. It thus
```

facilitates data sharing, eases metadata querying and enables automatic data
analysis pipelines.

While BIDS was originally intended for Magnetic Resonance Imaging (MRI)
data, it has recently been extended for electro-/magneto-encephalography (EEG/MEG)
[@Niso2018, @Pernet2019], and intracranial EEG (iEEG) [@Holdgraf2019].
In this context, many software packages for electrophysiology data analysis are

> now becoming BIDS compatible. BIDS is a rich system to curate, aggregate and

Reviewer: In the current phrasing, this sentence jumps out of nowhere. I wanted to make the point that BIDS really allows you to do things more than what `.fif`, `.vhdr` etc can do -- and so you need a dedicated software that allows deeper integration rather than just compatibility. If you want to keep it like this, I think the sentence should be moved up with the rest of the stuff where we discuss what BIDS does.

> annotate neuroimaging databases. For this reason we developed a dedicated

Reviewer: ```suggestion
annotate neuroimaging databases. For this reason, we developed a dedicated
```

[BIDS validator](https://github.com/bids-standard/bids-validator).
Moreover, ``MNE-BIDS`` supports converting data formats that are not BIDS
compatible into permissible formats. As a result of this, users can easily convert
their datasets to BIDS in a matter of minutes, rather than hours of manual
labour.

In addition to this core functionality, ``MNE-BIDS`` is continuously being
extended to facilitate the analysis of BIDS formatted data.
To name some features, it is possible to read a BIDS dataset and obtain a
Python object, ready for analyis with ``MNE-Python``.
Users can save a T1-weighted anatomical MRI image alongside the MEG or EEG data
and apply an automatic defacing algorithm for anonymization purposes.
As a last example, anatomical landmark coordinates can be saved to enable coregistration
between the MEG/EEG and MRI data, which is necessary for
computation of forward and inverse solutions.

Users can easily install ``MNE-BIDS`` on all major platforms via `pip` and
`conda`, and its functionality is continuously tested on Windows, OS X, and
Linux.
Other than the core dependencies for scientific computing
(`numpy`, `scipy`) and handling of MEG/EEG data (`mne`),

> ``MNE-BIDS`` has minimal dependencies, all of which are optional. The Application

Reviewer 3: what do you have in mind for optional deps? I see only nibabel. If would rather be explicit that it needs numpy, scipy, mne and optionally nibabel.

Reviewer: agreed!

Reviewer 2: I think he might have also meant `pybv` as soon as that lands

Author: and then also `nilearn` for `from nilearn.plotting import plot_anat`, which we use in an example to show how plotting coregistered fiducials works.

And `scikit-learn`, because it is a dependency of `nilearn` :-)

I'll take a proposal if someone has a nice way to put this in a sentence.

Programming Interface (API) of
the package is stable and extensively documented and explained in examples
(https://mne.tools/mne-bids/). In addition, a command-line interface is provided
that allows even non-Python users to benefit from some of the core functionality.

As of writing, ``MNE-BIDS`` has received code contributions from 15
contributors and its user base is steadily growing. Code development is

> active](https://github.com/mne-tools/mne-bids/graphs/commit-activity) and the

Reviewer 3: ```suggestion
[active](https://github.com/mne-tools/mne-bids/graphs/commit-activity) and the
```

information between the electrophysiology and MRI data, which allows for
automatic computation of forward and inverse solutions.

Users can easily install ``MNE-BIDS`` on all platforms via `pip` and `conda`,
and its functionality is continuously tested on Windows and Linux.
Other than three core dependencies for scientific computing
(`numpy`, `scipy`) and handling of neurophysiological data (`mne`),
``MNE-BIDS`` has minimal dependencies, all of which are optional. The API of
the package is stable and extensively documented and explained in examples
(https://mne.tools/mne-bids/).

As of writing, ``MNE-BIDS`` has received code contributions from 15
contributors and its user base is steadily growing. Code development is
active and users are typically receiving support within a few hours of opening
an issue on our dedicated issue tracker.

MNE-BIDS is used as a dependency in several other software packages such as
the [MNE-study-template](https://github.com/mne-tools/mne-study-template), an
automated analysis pipeline based on [@Mainak2018], and
[Biscuit](https://github.com/Macquarie-MEG-Research/Biscuit), a graphical

> application to format BIDS data.

Reviewer: ```suggestion
user interface to format BIDS data.
```

Lastly, several large institutions have adopted MNE-BIDS for their
workflows such as the Martinos Center for Biomedical Imaging.

The developer team is excited to improve the state of the art in data handling
and looking forward to welcoming new contributors and users.

# Acknowledgements

MNE-BIDS development is partly supported by the Academy of Finland

> (grant 310988), and the Google Summer of Code 2019.

Reviewer 6: ```suggestion
(grant 310988), NIH NINDS R01-NS104585, and the Google Summer of Code 2019.
```

> (grant 310988), NIH NINDS R01-NS104585, and the Google Summer of Code 2019.

Reviewer 3: ```suggestion
(grant 310988), NIH NINDS R01-NS104585, ERC Starting Grant SLAB ERC-YStG-676943, ANR meegBIDS.fr and the Google Summer of Code 2019.
```

With the emergence of the Brain Imaging Data Structure
(``BIDS``; [@Gorgolewski2016]) in 2016, the neuroscientific community received
a standard to organize and share data. Originally limited to MRI data formats,
BIDS is continuously being extended and by now also supports other neuroimaging
modalities such as MEG [@Niso2018], EEG [@Pernet2019], and iEEG [@Holdgraf2019].

BIDS prescribes how complex experimental data should be structured and which
metadata should be present besides the raw data. This set of guidelines
provides the following benefits:

1. **Sharing data within a lab and between labs:** The BIDS specification
   empowers users to understand any BIDS dataset without requiring specialist
   knowledge.
1. **Data validation:** Dedicated software can check whether the rules of
   BIDS are being followed in a dataset and alert users to missing or broken
   data (see
   [bids-validator](https://github.com/bids-standard/bids-validator)).
1. **Automatic data handling and analysis pipelines:** Knowing which data to
   expect and where to search for it, many common workflows in data handling
   and analysis can be simplified or automated through specialized software
   tools.

> To make use of these benefits, datasets must first be converted to the

Reviewer: I am not sure "conversion" is the right phrase that we want to advertise here. The real power of `mne-bids` is in the ability to interface with MNE seamlessly. BIDS is a complex (yet standardized) data structure that needs specialized software. It's not that `mne-bids` will be useless once all datasets are converted to BIDS format because you still need specialized tools to organize data in BIDS format and to understand them in MNE

As a last example, fiducial points can be saved to store coregistration
information between the electrophysiology and MRI data, which allows for
automatic computation of forward and inverse solutions.

Users can easily install ``MNE-BIDS`` on all majorplatforms via `pip` and
`conda`, and its functionality is continuously tested on Windows, OSX, and
Linux.
Other than three core dependencies for scientific computing
(`numpy`, `scipy`) and handling of neurophysiological data (`mne`),
``MNE-BIDS`` has minimal dependencies, all of which are optional. The API of
the package is stable and extensively documented and explained in examples
(https://mne.tools/mne-bids/).

As of writing, ``MNE-BIDS`` has received code contributions from 15
contributors and its user base is steadily growing. Code development is
active (https://github.com/mne-tools/mne-bids/graphs/commit-activity) and the
developer team is committed to provide timely support for issues opened on the
GitHub issue tracker.

MNE-BIDS is used as a dependency in several other software packages such as
the [MNE-study-template](https://github.com/mne-tools/mne-study-template), an

> automated analysis pipeline based on [@Mainak2018], and

Reviewer: ```suggestion
automated pipeline for group analysis with MNE [@Mainak2018], and
```

[Biscuit](https://github.com/Macquarie-MEG-Research/Biscuit), a graphical
user interface to format BIDS data.

> Lastly, several large institutions have adopted MNE-BIDS for their

Reviewer: @alexrockhill can you confirm this?

Author: I was basing this off an email that an employee requested MNE-BIDS to be allowed in the Martinos Python distribution. (Email from 3rd of June)

- index: 5
  name: Human Cortical Physiology and Neurorehabilitation Section, NINDS, NIH, Bethesda, Maryland 20892
- index: 6
  name: UC Berkeley, Project Jupyter
- index: 7
  name: Institut du cerveau et de la moelle épinière (ICM), Paris, France
- index: 8
  name: Department of Biomedical and Clinical Sciences 'L. Sacco', University of Milan, Milan, Italy
- index: 9
  name: Institute for Learning and Brain Sciences, University of Washington, Seattle, WA, USA
- index: 10
  name: Institute of Neuroscience and Medicine (INM-3), Research Center Jülich, Germany
- index: 11
  name: Max-Planck-Institute for Empirical Aesthetics, Frankfurt a.M., Germany
- index: 12
  name: Institute of Psychology, University of Graz, Austria
- index: 13
  name: University of Oregon, Eugene OR, USA
- index: 14
  name: Université Paris-Saclay, Inria, CEA, Palaiseau, France
- index: 15
  name: Athinoula A. Martinos Center for Biomedical Imaging, Massachusetts General Hospital, Charlestown, MA, USA
date: 4 November 2019
bibliography: paper.bib
---

# Summary

With the development of the Brain Imaging Data Structure

> [``BIDS``; @Gorgolewski2016], the neuroscientific community received

Reviewer 4: ```suggestion
[BIDS; @Gorgolewski2016], the neuroscientific community received
```

Reviewer 4: In other places, we format BIDS as a normal acronym.

# Summary

With the development of the Brain Imaging Data Structure
(``BIDS``; [@Gorgolewski2016]), the neuroscientific community received
a standard to organize and share data. BIDS standardizes storage of complex experimental
data by prescribing a set of file naming conventions and templates to store the
associated metadata in the form of Javascript Object Notation (JSON) files. It thus
facilitates data sharing, eases metadata querying and enables automatic data
analysis pipelines.

While BIDS was originally intended for Magnetic Resonance Imaging (MRI)
data, it has recently been extended for electro-/magneto-encephalography (EEG/MEG)

> [@Niso2018, @Pernet2019], and intracranial EEG (iEEG) [@Holdgraf2019].

Reviewer 2: with this order of electro/magneto, you should reorder the references. better yet, you might just want to break it up and just say magneto..., electro..., intracranial...

In this context, many software packages for electrophysiology data analysis are
now becoming BIDS compatible. BIDS is a rich system to curate, aggregate and
annotate neuroimaging databases. For this reason we developed a dedicated
software, ``MNE-BIDS`` with the goal to provide

> a programmable interface for BIDS datasets in electrophysiology. ``MNE-BIDS`` allows

Reviewer 2: i think elsewhere, you've used neurophysiology as the catchall for all three

Reviewer: what's the difference? :)
